core/src/services/swift/lister.rs

```diff
@@ -96,9 +96,14 @@ impl oio::PageList for SwiftLister {
                     meta.set_content_md5(hash.as_str());
 
                     // we'll change "2023-10-28T19:18:11.682610" to "2023-10-28T19:18:11.682610Z"
-                    last_modified.push('Z');
+                    if !last_modified.ends_with('Z') {
```

Comment:
Would you like to add more context about this? Which service will return `last_modified` without `Z`?

Reply:
Okay, I've added an explanation here that OpenStack Swift can cause errors.